src/Http/Client/Adapter/Stream.php

```diff
@@ -88,7 +88,7 @@ public function send(RequestInterface $request, array $options): array
      * Creates one or many response objects based on the number
      * of redirects that occurred.
      *
-     * @param array<string> $headers The list of headers from the request(s)
+     * @param list<string> $headers The list of headers from the request(s)
```

Comment:
We are not using list anymore.

Reply:
For this particular method the argument must be a list else it can cause errors. It collects indexes of the $hoarders argument and does `isset(index +1)` checks.